src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,185 @@
+    private static final String NORMAL_LOGIN = "alexvolodin@gmail.com";
+    private static final String NORMAL_PASSWORD = "volodinalex10";
+    private static final int NORMAL_AGE = 23;
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        storageDao.clearStorage();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(InvalidDataException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_validUser_isOkay() {
+        User validUser = new User(NORMAL_LOGIN, NORMAL_PASSWORD, NORMAL_AGE);
+        User registered = assertDoesNotThrow(() -> registrationService.register(validUser));
+
+        assertEquals(storageDao.get(validUser.getLogin()), registered);
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User userNullLogin = new User(null, NORMAL_PASSWORD, NORMAL_AGE);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(userNullLogin));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User userNullPassword = new User(NORMAL_LOGIN, null, NORMAL_AGE);
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(userNullPassword));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User userNullAge = new User(NORMAL_LOGIN, NORMAL_PASSWORD, null);
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(userNullAge));
+    }
+
+    @Test
+    void register_existedUser_notOk() {
+        User existedUser = new User(NORMAL_LOGIN, NORMAL_PASSWORD, NORMAL_AGE);
+        storageDao.add(existedUser);
+
+        User newUser = new User(NORMAL_LOGIN, "alexNewPassword", 26);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(newUser));
+    }
+
+    @Test
+    void register_lowLengthLogin_notOk() {
+        User user1LowLengthLogin = new User("", NORMAL_PASSWORD, NORMAL_AGE);
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user1LowLengthLogin));
+
+        User user2LowLengthLogin = new User("ale", NORMAL_PASSWORD, NORMAL_AGE);
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user2LowLengthLogin));
+
+        User user3LowLengthLogin = new User("alexa", NORMAL_PASSWORD, NORMAL_AGE);
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(user3LowLengthLogin));
+    }
+
+    @Test
+    void register_validLengthLogin_isOkay() {
+        User user1ValidLengthLogin = new User("alex10", NORMAL_PASSWORD, NORMAL_AGE);
+        User registered1 = assertDoesNotThrow(
```

Comment:
same for other places
```suggestion
        User registered1 = 
```